src/opnsense/service/templates/OPNsense/IDS/suricata.yaml

```diff
@@ -923,9 +926,10 @@ app-layer:
       detection-ports:
         dp: 443
 
-      # Generate JA3 fingerprint from client hello. If not specified it
+      # Generate JA3/JA4 fingerprints from client hello. If not specified it
       # will be disabled by default, but enabled if rules require it.
       ja3-fingerprints: yes
+      ja4-fingerprints: yes
```

Comment:
question is if we actually need to enable this, unless no rules actually use the option in your use-case. There might be a performance penalty, so just wanted to double check.

Reply:
I've enabled it given it behaves like JA3. From the [documentation](https://docs.suricata.io/en/latest/output/eve/eve-json-format.html#id10):
> JA3 and JA4 must be enabled in the Suricata config file (set 'app-layer.protocols.tls.ja3-fingerprints'/'app-layer.protocols.tls.ja4-fingerprints' to 'yes').

Technically, we could for each JA3/JA4:
- Set it to `yes` if either extended TLS logging is enabled or if the field is explicitly mentioned in the custom TLS fields.
- Not specify it otherwise, letting Suricata determine whether it is required if a rule needs it.

I however did not take this approach as it is technically a "breaking" change for JA3 (i.e., changing the default).